recirq/quantum_chess/quantum_board.py

```diff
@@ -775,624 +779,599 @@ def do_move_internal(self, m: move.Move) -> int:
             for val in list(self.allowed_pieces):
                 self.allowed_pieces.add(val - 1)
 
-        if m.move_type == enums.MoveType.PAWN_EP:
-            # For en passant, first determine the square of the pawn being
-            # captured, which should be next to the target.
-            if m.target[1] == "6":
-                epbit = square_to_bit(m.target[0] + "5")
-            elif m.target[1] == "3":
-                epbit = square_to_bit(m.target[0] + "4")
-            else:
-                raise ValueError(f"Invalid en passant target {m.target}")
-            epqubit = bit_to_qubit(epbit)
+        measurement = move_func[m.move_type](m, sbit, tbit, squbit, tqubit)
+        assert measurement in (0, 1)
+        if self.reset_starting_states and initial_circuit != self.circuit:
+            self.reset_if_classical()
+        return measurement
+
+    def _do_move_pawn_ep(self, m, sbit, tbit, squbit, tqubit):
```

Comment:
We should add type info to this function and others.  Docstrings would be nice too.

Reply:
Do you really think it would help? For me the key to understanding these functions is to read the place in do_move where the function table is used; having type and description boilerplate on each one wouldn't really help.

Reply:
It would help in typechecking the contents of the function I suppose.  It's also possible to change the function table to be typed as a Dict[str,Callable[arg,types,here]] so that you could make sure that every function in the table has the same type.  

I think docstrings would be nice, especially for complicated functions so that various implementation details (like when  qubits are unhooked or whatever can be explained).  Also nice to have something to explain that "_ep" means en passant, for instance.